gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2

```diff
@@ -139,26 +139,39 @@ def _get_http_options():
     return response
 {% endmacro %}
 
-{% macro rest_call_method_common(body_spec, method_name, service_name) %}
+{% macro rest_call_method_common(body_spec, method_name, service_name, is_async=False) %}
+{% set await_prefix = "await " if is_async else "" %}
+{% set async_prefix = "async " if is_async else "" %}
+{% set async_class_prefix = "Async" if is_async else "" %}
 
     http_options = _Base{{ service_name }}RestTransport._Base{{method_name}}._get_http_options()
+    {% if not is_async %}
     request, metadata = self._interceptor.pre_{{ method_name|snake_case }}(request, metadata)
+    {% endif %}
     transcoded_request = _Base{{ service_name }}RestTransport._Base{{method_name}}._get_transcoded_request(http_options, request)
 
     {% if body_spec %}
     body = _Base{{ service_name }}RestTransport._Base{{method_name}}._get_request_body_json(transcoded_request)
-    {% endif %} {# body_spec #}
+    {% endif %}{# body_spec #}
 
     # Jsonify the query params
     query_params = _Base{{ service_name }}RestTransport._Base{{method_name}}._get_query_params_json(transcoded_request)
 
     # Send the request
-    response = {{ service_name }}RestTransport._{{method_name}}._get_response(self._host, metadata, query_params, self._session, timeout, transcoded_request{% if body_spec %}, body{% endif %})
+    response = {{ await_prefix }}{{ async_class_prefix }}{{ service_name }}RestTransport._{{method_name}}._get_response(self._host, metadata, query_params, self._session, timeout, transcoded_request{% if body_spec %}, body{% endif %})
 
     # In case of error, raise the appropriate core_exceptions.GoogleAPICallError exception
     # subclass.
     if response.status_code >= 400:
+        {% if is_async %}
```

Comment:
Add a comment to clarify the reason that `format_http_response_error` has more parameters for async compared to sync. If the sync code will be cleaned up later, add a TODO comment with a link to a bug

Reply:
Addressed.